The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6472377745/job/17572899928) [(plain)](https://github.com/rust-lang/rust/commit/01f025cc79dde36937100d2f7996748f3201f341/checks/17572899928/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=ChrisDenton
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_e8a3c42a-7744-4d0d-ac99-b3a1230d60ee
GITHUB_EVENT_NAME=pull_request
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=a79ab2725a5de73a36b972ad2ac380aef9723e0b
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_e8a3c42a-7744-4d0d-ac99-b3a1230d60ee
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_e8a3c42a-7744-4d0d-ac99-b3a1230d60ee
GITHUB_TRIGGERING_ACTOR=ChrisDenton
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/116606/merge
GITHUB_WORKFLOW_SHA=a79ab2725a5de73a36b972ad2ac380aef9723e0b
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_19_X64=/opt/hostedtoolcache/go/1.19.13/x64
---
    Checking addr2line v0.21.0
error[E0308]: mismatched types
    --> library/std/src/sys/windows/fs.rs:1075:49
     |
1075 |         return Err(io::Error::from_raw_os_error(c::ERROR_PATH_NOT_FOUND));
     |                    ---------------------------- ^^^^^^^^^^^^^^^^^^^^^^^ expected `i32`, found `u32`
     |                    arguments to this function are incorrect
     |
note: associated function defined here
    --> library/std/src/io/error.rs:622:12
    --> library/std/src/io/error.rs:622:12
     |
622  |     pub fn from_raw_os_error(code: RawOsError) -> Error {
     |            ^^^^^^^^^^^^^^^^^ ----------------
help: you can convert a `u32` to an `i32` and panic if the converted value doesn't fit
     |
1075 |         return Err(io::Error::from_raw_os_error(c::ERROR_PATH_NOT_FOUND.try_into().unwrap()));

For more information about this error, try `rustc --explain E0308`.
error: could not compile `std` (lib) due to previous error
Build completed unsuccessfully in 0:00:12
```

</details>

--- a/library/std/src/sys/windows/fs.rs
+++ b/library/std/src/sys/windows/fs.rs
@@ -1066,6 +1066,14 @@ impl DirBuilder {
 }
 
 pub fn readdir(p: &Path) -> io::Result<ReadDir> {
+    // We push a `*` to the end of the path which cause the empty path to be
+    // treated as the current directory. So, for consistency with other platforms,
+    // we explicitly error on the empty path.
+    if p.as_os_str().is_empty() {
+        // Return an error code consistent with other ways of opening files.
+        // E.g. fs::metadata or File::open.
+        return Err(io::Error::from_raw_os_error(c::ERROR_PATH_NOT_FOUND as i32));
+    }
     let root = p.to_path_buf();
     let star = p.join("*");
     let path = maybe_verbatim(&star)?;

